### ISSUE TITLE
conda env create/update --json broken in 24.x

### ISSUE THREAD

**Issue by beenje** [conda/conda#13835]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

The `--json` output flag doesn't work anymore with 24.3.0 (seems broken since 24.1).

With the following `environment.yml`:

```yaml
name: foo
channels:
  - conda-forge
dependencies:
  - pip=24.0
  - python=3.12.2
  - pip:
    - httpx==0.27.0
```

Running ` conda env create -f environment.yml --json` prints the same output as without `--json`.

```bash
$ conda env create -f environment.yml --json
Installing pip dependencies: | Ran pip subprocess with arguments:
['/opt/conda/envs/foo/bin/python', '-m', 'pip', 'install', '-U', '-r', '/build/condaenv.rzr9udv6.requirements.txt', '--exists-action=b']
Pip subprocess output:
Collecting httpx==0.27.0 (from -r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)
Collecting anyio (from httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached anyio-4.3.0-py3-none-any.whl.metadata (4.6 kB)
Collecting certifi (from httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached certifi-2024.2.2-py3-none-any.whl.metadata (2.2 kB)
Collecting httpcore==1.* (from httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached httpcore-1.0.5-py3-none-any.whl.metadata (20 kB)
Collecting idna (from httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached idna-3.7-py3-none-any.whl.metadata (9.9 kB)
Collecting sniffio (from httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
Collecting h11<0.15,>=0.13 (from httpcore==1.*->httpx==0.27.0->-r /build/condaenv.rzr9udv6.requirements.txt (line 1))
  Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
Using cached httpx-0.27.0-py3-none-any.whl (75 kB)
Using cached httpcore-1.0.5-py3-none-any.whl (77 kB)
Using cached anyio-4.3.0-py3-none-any.whl (85 kB)
Using cached idna-3.7-py3-none-any.whl (66 kB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
Using cached h11-0.14.0-py3-none-any.whl (58 kB)
Installing collected packages: sniffio, idna, h11, certifi, httpcore, anyio, httpx
Successfully installed anyio-4.3.0 certifi-2024.2.2 h11-0.14.0 httpcore-1.0.5 httpx-0.27.0 idna-3.7 sniffio-1.3.1

done
#
# To activate this environment, use
#
#     $ conda activate foo
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

Same with `conda env update -f environment.yml --json`:

```bash
$ conda env update -f environment.yml --json
Channels:
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done
Installing pip dependencies: \ Ran pip subprocess with arguments:
['/opt/conda/envs/foo/bin/python', '-m', 'pip', 'install', '-U', '-r', '/build/condaenv.lsa0g7gq.requirements.txt', '--exists-action=b']
Pip subprocess output:
Requirement already satisfied: httpx==0.27.0 in /opt/conda/envs/foo/lib/python3.12/site-packages (from -r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (0.27.0)
Requirement already satisfied: anyio in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (4.3.0)
Requirement already satisfied: certifi in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (2024.2.2)
Requirement already satisfied: httpcore==1.* in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (1.0.5)
Requirement already satisfied: idna in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (3.7)
Requirement already satisfied: sniffio in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (1.3.1)
Requirement already satisfied: h11<0.15,>=0.13 in /opt/conda/envs/foo/lib/python3.12/site-packages (from httpcore==1.*->httpx==0.27.0->-r /build/condaenv.lsa0g7gq.requirements.txt (line 1)) (0.14.0)

done
#
# To activate this environment, use
#
#     $ conda activate foo
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

With conda 23.11.0, it's working as expected:

```bash
$ conda env create -f environment.yml --json 2>/dev/null
{
  "actions": {
    "FETCH": [],
    "LINK": [
      {
        "base_url": "https://conda.anaconda.org/conda-forge",
        "build_number": 0,
        "build_string": "conda_forge",
        "channel": "conda-forge",
        "dist_name": "_libgcc_mutex-0.1-conda_forge",
        "name": "_libgcc_mutex",
        "platform": "linux-64",
        "version": "0.1"
      },
 ...
      {
        "base_url": "https://conda.anaconda.org/conda-forge",
        "build_number": 0,
        "build_string": "pyhd8ed1ab_0",
        "channel": "conda-forge",
        "dist_name": "pip-24.0-pyhd8ed1ab_0",
        "name": "pip",
        "platform": "noarch",
        "version": "24.0"
      }
    ],
    "PIP": [
      "anyio-4.3.0",
      "certifi-2024.2.2",
      "h11-0.14.0",
      "httpcore-1.0.5",
      "httpx-0.27.0",
      "idna-3.7",
      "sniffio-1.3.1"
    ],
    "PREFIX": "/opt/conda/envs/foo"
  },
  "prefix": "/opt/conda/envs/foo",
  "success": true
}

$ conda env update -f environment.yml --json 2>/dev/null
{
  "message": "All requested packages already installed.",
  "success": true
}
```

### Conda Info

```shell
active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /root/.condarc
 populated config files : /opt/conda/.condarc
          conda version : 24.3.0
    conda-build version : not installed
         python version : 3.10.14.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=x86_64_v4
                          __conda=24.3.0=0
                          __glibc=2.31=0
                          __linux=5.15.49=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/24.3.0 requests/2.31.0 CPython/3.10.14 Linux/5.15.49-linuxkit ubuntu/20.04.6 glibc/2.31 solver/libmamba conda-libmamba-solver/24.1.0 libmambapy/1.5.8
                UID:GID : 0:0
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /opt/conda/.condarc <==
channels:
  - conda-forge
```

### Conda list

```shell
# packages in environment at /opt/conda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
archspec                  0.2.3              pyhd8ed1ab_0    conda-forge
boltons                   24.0.0             pyhd8ed1ab_0    conda-forge
brotli-python             1.1.0           py310hc6cd4ac_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.28.1               hd590300_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py310h2fee648_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     24.3.0          py310hff52083_0    conda-forge
conda-libmamba-solver     24.1.0             pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
distro                    1.9.0              pyhd8ed1ab_0    conda-forge
fmt                       10.2.1               h00ab1b0_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
jsonpatch                 1.33               pyhd8ed1ab_0    conda-forge
jsonpointer               2.4             py310hff52083_3    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libarchive                3.7.2                h2aa1ff5_1    conda-forge
libcurl                   8.7.1                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libmamba                  1.5.8                had39da4_0    conda-forge
libmambapy                1.5.8           py310h39ff949_0    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsolv                   0.7.28               hfc55251_2    conda-forge
libsqlite                 3.45.2               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_5    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxml2                   2.12.6               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mamba                     1.5.8           py310h51d5547_0    conda-forge
menuinst                  2.0.2           py310hff52083_0    conda-forge
ncurses                   6.4.20240210         h59595ed_0    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
packaging                 24.0               pyhd8ed1ab_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
platformdirs              4.2.0              pyhd8ed1ab_0    conda-forge
pluggy                    1.4.0              pyhd8ed1ab_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.6           py310h2372a71_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.14         hd12c33a_0_cpython    conda-forge
python_abi                3.10                    4_cp310    conda-forge
readline                  8.2                  h8228510_1    conda-forge
reproc                    14.2.4.post0         hd590300_1    conda-forge
reproc-cpp                14.2.4.post0         h59595ed_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
ruamel.yaml               0.18.6          py310h2372a71_0    conda-forge
ruamel.yaml.clib          0.2.8           py310h2372a71_0    conda-forge
setuptools                69.5.1             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tqdm                      4.66.2             pyhd8ed1ab_0    conda-forge
truststore                0.8.0              pyhd8ed1ab_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml-cpp                  0.8.0                h59595ed_0    conda-forge
zstandard                 0.22.0          py310h1275a96_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

**Comment by beenje** [comment]
I had a quick Look and noticed that the [pip installer](https://github.com/conda/conda/blob/be5adf514550260952f20a807748f7f3ae297f10/conda/env/installers/pip.py#L74) isn't using the `args` Namespace but is taking the `json` flag from the `context`. Looks like the `context` isn't updated with the flags passed to the `conda env` command.

**Comment by travishathaway** [comment]
@beenje,

Thanks for reporting this. We actively looking at it.

**Comment by beeankha** [comment]
While investigating a solution to this regression issue, @kenodegard and I have found that setting the environment variable `CONDA_JSON=True` will result in correct/expected behavior. Due to this workaround, we're downgrading the severity to a https://github.com/conda/conda/labels/severity%3A%3A2; we're working on an actual fix that will be included in the 24.5.x release coming out next month.
